core/domain/skill_domain.py

```diff
@@ -653,3 +653,36 @@ def to_dict(self):
             'skill_model_created_on': self.skill_model_created_on,
             'skill_model_last_updated': self.skill_model_last_updated
         }
+
+
+class UserSkillMastery(object):
+    """Domain object for a user's mastery of a particular skill."""
+
+    def __init__(self, model_id, user_id, skill_id, degree_of_mastery):
+        """Constructs a SkillMastery domain object.
+
+        Args:
+            model_id: str. The model id corresponding to the user and
```

Comment:
I thought, in a previous review comment, I mentioned that the model_id should not be exposed to the domain layer. In particular, it should not be a field in the domain object since it has no meaning outside the datastore. Please fix.

Reply:
So later in the ```__init__``` function. Should I call the contruct_model_id function to store in the id?
Something like this: ```self.id = user_models.UserSkillMasteryModel.construct_model_id(user_id, skill_id)```?
Sorry, I'm a bit confused here.

Reply:
Yes, but that ID is not constructed from anything else. In this case what I'm saying is to not have the ID in the domain object at all, since user_id and skill_id already provide all the data required.

Reply:
okay